kapp-controller crashes when the there is an unsupported templating method specified in the App CRD

**What steps did you take:**
Create an App CRD as follows:

```
apiVersion: kappctrl.k14s.io/v1alpha1
kind: App
metadata:
  name: test
  namespace: default
spec:
  fetch:
    - image:
        url: some-image-with-files
  template:
    - foo: {} 
```

**What happened:**
Kapp-controller panics with a nil pointer deref:

```
E0106 15:08:52.526837      15 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 174 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14b6920, 0x236bbc0)
	k8s.io/apimachinery@v0.17.2/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.17.2/pkg/util/runtime/runtime.go:48 +0x82
panic(0x14b6920, 0x236bbc0)
	runtime/panic.go:679 +0x1b2
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).template(0xc00043d680, 0xc0003aefa0, 0x46, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_template.go:50 +0x56c
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).reconcileFetchTemplateDeploy(0xc00043d680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_reconcile.go:127 +0x48a
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).reconcileDeploy(0xc00043d680, 0x167d2f5, 0xe)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_reconcile.go:81 +0x1a6
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).Reconcile(0xc00043d680, 0xc000437400, 0x0, 0x0, 0x0)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_reconcile.go:44 +0x37f
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*CRDApp).Reconcile(...)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/crd_app.go:112
github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller.(*AppsReconciler).Reconcile(0xc00025e580, 0xc000806788, 0x7, 0xc000806784, 0x4, 0xc00000d9c0, 0xc00046bba0, 0x1344e55, 0xc000364888)
	github.com/vmware-tanzu/carvel-kapp-controller@/cmd/controller/apps_reconciler.go:38 +0x353
github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller.(*ErrReconciler).Reconcile(0xc00043b4c0, 0xc000806788, 0x7, 0xc000806784, 0x4, 0xc0008067f0, 0xc, 0xc00046bc30, 0x403744)
	github.com/vmware-tanzu/carvel-kapp-controller@/cmd/controller/err_reconciler.go:24 +0x177
github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller.(*UniqueReconciler).Reconcile(0xc000364870, 0xc000806788, 0x7, 0xc000806784, 0x4, 0xc00046bcd8, 0xc0000d4360, 0xc0000d42d8, 0x186fd60)
	github.com/vmware-tanzu/carvel-kapp-controller@/cmd/controller/unique_reconciler.go:40 +0x19e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00003a180, 0x15081c0, 0xc00000d840, 0x0)
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00003a180, 0x0)
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00003a180)
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0002be280)
	k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0002be280, 0x3b9aca00, 0x0, 0x1, 0xc00021c000)
	k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0002be280, 0x3b9aca00, 0xc00021c000)
	k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:193 +0x328
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x11ec9bc]

goroutine 174 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.17.2/pkg/util/runtime/runtime.go:55 +0x105
panic(0x14b6920, 0x236bbc0)
	runtime/panic.go:679 +0x1b2
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).template(0xc00043d680, 0xc0003aefa0, 0x46, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_template.go:50 +0x56c
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).reconcileFetchTemplateDeploy(0xc00043d680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_reconcile.go:127 +0x48a
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).reconcileDeploy(0xc00043d680, 0x167d2f5, 0xe)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_reconcile.go:81 +0x1a6
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*App).Reconcile(0xc00043d680, 0xc000437400, 0x0, 0x0, 0x0)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/app_reconcile.go:44 +0x37f
github.com/vmware-tanzu/carvel-kapp-controller/pkg/app.(*CRDApp).Reconcile(...)
	github.com/vmware-tanzu/carvel-kapp-controller@/pkg/app/crd_app.go:112
github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller.(*AppsReconciler).Reconcile(0xc00025e580, 0xc000806788, 0x7, 0xc000806784, 0x4, 0xc00000d9c0, 0xc00046bba0, 0x1344e55, 0xc000364888)
	github.com/vmware-tanzu/carvel-kapp-controller@/cmd/controller/apps_reconciler.go:38 +0x353
github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller.(*ErrReconciler).Reconcile(0xc00043b4c0, 0xc000806788, 0x7, 0xc000806784, 0x4, 0xc0008067f0, 0xc, 0xc00046bc30, 0x403744)
	github.com/vmware-tanzu/carvel-kapp-controller@/cmd/controller/err_reconciler.go:24 +0x177
github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller.(*UniqueReconciler).Reconcile(0xc000364870, 0xc000806788, 0x7, 0xc000806784, 0x4, 0xc00046bcd8, 0xc0000d4360, 0xc0000d42d8, 0x186fd60)
	github.com/vmware-tanzu/carvel-kapp-controller@/cmd/controller/unique_reconciler.go:40 +0x19e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00003a180, 0x15081c0, 0xc00000d840, 0x0)
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00003a180, 0x0)
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00003a180)
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0002be280)
	k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0002be280, 0x3b9aca00, 0x0, 0x1, 0xc00021c000)
	k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0002be280, 0x3b9aca00, 0xc00021c000)
	k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:193 +0x328
{"level":"error","ts":1609945732.5325596,"logger":"kc.init","msg":"Could not start controller","error":"exit status 2","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tgithub.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/vmware-tanzu/carvel-kapp-controller/cmd/controllerinit.Run\n\tgithub.com/vmware-tanzu/carvel-kapp-controller@/cmd/controllerinit/run.go:36\nmain.main\n\tgithub.com/vmware-tanzu/carvel-kapp-controller@/cmd/main.go:43\nruntime.main\n\truntime/proc.go:203"}
```

--- a/pkg/app/app_template.go
+++ b/pkg/app/app_template.go
@@ -35,7 +35,7 @@ func (a *App) template(dirPath string) exec.CmdRunResult {
 			template = a.templateFactory.NewSops(*tpl.Sops, genericOpts)
 		default:
 			result.AttachErrorf("%s", fmt.Errorf("Unsupported way to template"))
-			break
+			return result
 		}
 
 		if isStream {

